unison-cli/src/Unison/Cli/MonadUtils.hs

```diff
@@ -474,8 +474,8 @@ updateRoot new reason =
     let newHash = Branch.headHash new
     oldHash <- getLastSavedRootHash
     when (oldHash /= newHash) do
-      setRootBranch new
```

Comment:
Moving the memory-branch update until AFTER sqlite is updated means the new branch will actually be there when the LSP goes to load it.